src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -5159,6 +5177,24 @@ void Assembler::evpmadd52luq(XMMRegister dst, KRegister mask, XMMRegister src1,
   emit_int16((unsigned char)0xB4, (0xC0 | encode));
 }
 
+void Assembler::vpmadd52huq(XMMRegister dst, XMMRegister src1, Address src2, int vector_len) {
+  assert(vector_len == AVX_512bit ? VM_Version::supports_avx512ifma() : VM_Version::supports_avxifma(), "");
+  InstructionMark im(this);
+  InstructionAttr attributes(vector_len, /* rex_w */ true, /* legacy_mode */ false, /* no_mask_reg */ false, /* uses_vl */ true);
+
+  vex_prefix(src2, src1->encoding(), dst->encoding(), VEX_SIMD_66, VEX_OPCODE_0F_38, &attributes);
+  emit_int8((unsigned char)0xB5);
+  emit_operand(dst, src2, 0);
+}
+
+void Assembler::vpmadd52huq(XMMRegister dst, XMMRegister src1, XMMRegister src2, int vector_len) {
+  assert(vector_len == AVX_512bit ? VM_Version::supports_avx512ifma() : VM_Version::supports_avxifma(), "");
```

Comment:
Same as above.

Reply:
Please see the updated assert as suggested for vpmadd52[l/h]uq in the latest commit.